frontend/src/app/json-editor/_components/Note.tsx

```diff
@@ -49,6 +51,12 @@ function Note({
     setIsValid(result.success);
     setZodErrors(result.success ? null : result.error);
     setScript(updated);
+    // Restore focus after state update
+    if (key === "text") {
+      setTimeout(() => {
```

Comment:
Why does the input field loose focus in the first place? Would it not be better to fix the underlying issue?

Reply:
I don’t know why it’s behaving that way. I’ve tried debugging the issue but haven’t found a solution. This will have to do for now. The page is only used 1–2 times a day, so it’s not worth spending hours debugging it.